### PR TITLE
support regexp config

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Provide `regexp` config type so user can do for example:
```
...
filter    config.Regexp
...
```

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
As commented https://github.com/lyft/flytepropeller/pull/90#discussion_r393109951, it is better to have an dedicated `regexp` config type so validation can be done properly and early on.

https://github.com/lyft/flytepropeller/pull/90 Might not be needed after all, but having such a type may still make sense.

## Tracking Issue
NA

## Follow-up issue
NA

### CODE REVIEW

**Comment by kumare3** [comment]
Ohh wow, never saw this. Lets ship it!

**Comment by honnix** [comment]
Thanks for the approval. I forgot this PR too. :)
